flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -149,7 +156,7 @@ public void snapshotState(StateSnapshotContext context) throws Exception {
     LOG.info("Start to flush snapshot state to state backend, table: {}, checkpointId: {}", table, checkpointId);
 
     // Update the checkpoint state.
-    dataFilesPerCheckpoint.put(checkpointId, ImmutableList.copyOf(dataFilesOfCurrentCheckpoint));
+    dataFilesPerCheckpoint.put(checkpointId, writeToManifest(checkpointId));
```

Comment:
Since we are now writing manifest to state, do we need to rename datafiles related variables to manifest context?

Reply:
Make sense !